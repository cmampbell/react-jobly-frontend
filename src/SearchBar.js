import React, { useState } from 'react';
import { Form } from 'react-router-dom';

const SearchBar = ({path}) => {
    const INITIAL_STATE = '';
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (evt) => {
        setFormData(() => evt.target.value)
    }

    return (
        <Form method="get" action={`/${path}`} onSubmit={(()=> setFormData(INITIAL_STATE))}>
            {path === 'companies' ? 
                        <input id="name" name="name" onChange={handleChange} value={formData} type='text'/>
                        : <input id="title" name="title" onChange={handleChange} value={formData} type='text'/>}
            <button type="submit">Search {path}</button>
        </Form>
    )
}

export default SearchBar;