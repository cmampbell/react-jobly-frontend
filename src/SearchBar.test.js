import { render } from '@testing-library/react';
import SearchBar from './SearchBar';
import { RouterProvider, createMemoryRouter} from 'react-router-dom';

const route = [
    {
      path: "/",
      element: <SearchBar path={'companies'} />,
    },
  ];

  const router = createMemoryRouter(route);

describe('Smoke and snapshot CompanyCard test', () => {
    it('should render without crashing', () => {
        render(<RouterProvider router={router}/>);
    });
    it('should match snapshot', ()=> {
        const { asFragment } = render(<RouterProvider router={router}/>);
        expect(asFragment()).toMatchSnapshot();
    })
})