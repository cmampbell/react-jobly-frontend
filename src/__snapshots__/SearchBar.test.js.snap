// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot CompanyCard test should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="/companies"
    method="get"
  >
    <input
      id="name"
      name="name"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search companies
    </button>
  </form>
</DocumentFragment>
`;
